### ISSUE TITLE
typed-property-needs-bind-usage

### ISSUE THREAD

**Issue by HeikoStudt** [vuejs/eslint-plugin-vue#1902]
**Please describe what the rule should do:**
Assuming you have defined a Prop as a Boolean or Number and use it in the component's template like in a ```v-if="isToShow"```.
Then, on usage, you must not use the string "false" but the value false as ```"false" != false``` or ommit the flag entirely.

This problem exists for Numbers as well. However, it is not that problematic as these are implicit converted in most of their usages, so long as one is not using some type-strict equal (as in the example).

**What category should the rule belong to?**
[ ] Enforces code style (layout)
[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

Example Component HelloWorld.vue
```vue
<template>
  <div>
    <span v-if="isToShown">Hello </span>
    <span v-if="countOfHellos !== 5">World</span>
  </div>
</template>

<script setup lang="ts">
const props = defineProps({
  isToShown: Boolean,
  countOfHellos : Number,
});
</script>
```

Example usages where the LINT rule will apply.
```vue
<template>
  <!-- (GOOD): (empty) -->
  <HelloWorld :isToShown="false" :countOfHellos="5" />

  <!-- (GOOD): (empty) -->
  <HelloWorld :countOfHellos="5" />

  <!-- (BAD): Hello -->
  <HW isToShown="false" :countOfHellos="5" />

  <!-- (BAD): World -->
  <HelloWorld :isToShown="false" countOfHellos="5" />
</template>
```

**Additional context**
Manages to find nearly invisible bugs introduced by missing colons. Reduces review & test time for this class of error.
Such a bug may be subtle as most such options if introduced may be used with ```true``` in mind and not quite tested setting them to ```false```. As you do not set the value of a flag to false explicitly to express the decision very often, you might not be aware that you set it to true-ish instead.

Note:
If you are using ```<script setup lang="js">``` instead (so no typescript), my IDE infers some better type off the defineProps and gives the message ```false is not a valid value for isToShown. Expected: isToShown``` which is at least some yellow warning. However, this does not name the issue correctly and currently this message is not shown for the number case and never in our lint's output where we can Ci it.

**Comment by HeikoStudt** [comment]
I do not know, whether the Linter does have the necessary information at the usage. However, at least https://github.com/vuejs/eslint-plugin-vue/issues/1870 and other prefer-order propositions seems to assume the information is available.
